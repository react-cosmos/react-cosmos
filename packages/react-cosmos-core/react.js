module.exports = require('./dist/react/index.js');
