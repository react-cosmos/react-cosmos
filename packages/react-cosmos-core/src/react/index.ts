export * from './getFixtureList.js';
export * from './getSortedDecoratorsForFixturePath.js';
export * from './types.js';
