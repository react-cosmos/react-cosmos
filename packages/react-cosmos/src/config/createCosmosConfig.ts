import path from 'path';
import { getCliArgs } from '../shared/cli';
import { getCosmosConfigPath } from './cosmosConfigPath';
import { resolveModule, resolvePath } from './resolve';
import { CosmosConfig, getRootDirAtPath } from './shared';

type CosmosConfigInput = Partial<CosmosConfig>;

export function createCosmosConfig(
  cosmosConfigInput: CosmosConfigInput = {}
): CosmosConfig {
  const cosmosConfigPath = getCosmosConfigPath();
  const rootDir = getRootDir(cosmosConfigInput, cosmosConfigPath);
  return {
    ...cosmosConfigInput,
    rootDir,
    exportPath: getExportPath(cosmosConfigInput, rootDir),
    staticPath: getStaticPath(cosmosConfigInput, rootDir),
    publicUrl: getPublicUrl(cosmosConfigInput),
    fixturesDir: getFixturesDir(cosmosConfigInput),
    fixtureFileSuffix: getFixtureFileSuffix(cosmosConfigInput),
    watchDirs: getWatchDirs(cosmosConfigInput, rootDir),
    userDepsFilePath: getUserDepsFilePath(cosmosConfigInput, rootDir),
    hostname: getHostname(cosmosConfigInput),
    port: getPort(cosmosConfigInput),
    globalImports: getGlobalImports(cosmosConfigInput, rootDir),
    ui: cosmosConfigInput.ui || {}
  };
}

function getRootDir({ rootDir }: CosmosConfigInput, cosmosConfigPath: string) {
  const cliArgs = getCliArgs();
  if (typeof cliArgs.rootDir === 'string') {
    return getRootDirAtPath(cliArgs.rootDir);
  }

  const configDir = path.dirname(cosmosConfigPath);
  return rootDir ? path.resolve(configDir, rootDir) : configDir;
}

function getExportPath(cosmosConfigInput: CosmosConfigInput, rootDir: string) {
  const { exportPath = 'cosmos-export' } = cosmosConfigInput;
  return resolvePath(rootDir, exportPath);
}

function getStaticPath(cosmosConfigInput: CosmosConfigInput, rootDir: string) {
  const { staticPath = null } = cosmosConfigInput;
  return staticPath && resolvePath(rootDir, staticPath);
}

function getPublicUrl({ publicUrl = '/' }: CosmosConfigInput) {
  return publicUrl;
}

function getFixturesDir({ fixturesDir = '__fixtures__' }: CosmosConfigInput) {
  return fixturesDir;
}

function getFixtureFileSuffix({
  fixtureFileSuffix = 'fixture'
}: CosmosConfigInput) {
  return fixtureFileSuffix;
}

function getWatchDirs(cosmosConfigInput: CosmosConfigInput, rootDir: string) {
  const { watchDirs = ['.'] } = cosmosConfigInput;
  return watchDirs.map(watchDir => resolvePath(rootDir, watchDir));
}

function getUserDepsFilePath(
  cosmosConfigInput: CosmosConfigInput,
  rootDir: string
) {
  const { userDepsFilePath = 'cosmos.userdeps.js' } = cosmosConfigInput;
  return resolvePath(rootDir, userDepsFilePath);
}

function getHostname({ hostname = null }: CosmosConfigInput) {
  return hostname;
}

function getPort(cosmosConfigInput: CosmosConfigInput) {
  const cliArgs = getCliArgs();
  if (typeof cliArgs.port === 'number') {
    return cliArgs.port;
  }

  const { port = 5000 } = cosmosConfigInput;
  return port;
}

function getGlobalImports(
  cosmosConfigInput: CosmosConfigInput,
  rootDir: string
) {
  const { globalImports = [] } = cosmosConfigInput;
  return globalImports.map(globalImport =>
    resolveModule(rootDir, globalImport)
  );
}
