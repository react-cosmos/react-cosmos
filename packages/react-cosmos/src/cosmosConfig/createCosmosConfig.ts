import path from 'path';
import { getCliArgs } from '../utils/cli.js';
import { resolveLoose } from '../utils/resolveLoose.js';
import {
  CosmosConfig,
  CosmosConfigInput,
  CosmosDomConfigInput,
} from './types.js';

export function createCosmosConfig(
  rootDir: string,
  cosmosConfigInput: CosmosConfigInput = {}
): CosmosConfig {
  return {
    ...cosmosConfigInput,
    rootDir,
    detectLocalPlugins: cosmosConfigInput.detectLocalPlugins ?? true,
    disablePlugins: cosmosConfigInput.disablePlugins ?? false,
    exposeImports: getExposeImports(cosmosConfigInput, rootDir),
    exportPath: getExportPath(cosmosConfigInput, rootDir),
    fixtureFileSuffix: getFixtureFileSuffix(cosmosConfigInput),
    fixturesDir: getFixturesDir(cosmosConfigInput),
    globalImports: getGlobalImports(cosmosConfigInput, rootDir),
    hostname: getHostname(cosmosConfigInput),
    https: getHttps(cosmosConfigInput),
    httpsOptions: getHttpsOptions(cosmosConfigInput, rootDir),
    ignore: getIgnore(cosmosConfigInput),
    lazy: getLazy(cosmosConfigInput),
    plugins: getPlugins(cosmosConfigInput, rootDir),
    port: getPort(cosmosConfigInput),
    portRetries: getPortRetries(cosmosConfigInput),
    publicUrl: cosmosConfigInput.publicUrl ?? '',
    rendererUrl: cosmosConfigInput.rendererUrl ?? null,
    staticPath: getStaticPath(cosmosConfigInput, rootDir),
    watchDirs: getWatchDirs(cosmosConfigInput, rootDir),
    dom: getDomConfig(cosmosConfigInput.dom || {}),
    ui: cosmosConfigInput.ui || {},
  };
}

function getExportPath(cosmosConfigInput: CosmosConfigInput, rootDir: string) {
  const { exportPath = 'cosmos-export' } = cosmosConfigInput;
  return path.resolve(rootDir, exportPath);
}

function getStaticPath(cosmosConfigInput: CosmosConfigInput, rootDir: string) {
  const { staticPath = null } = cosmosConfigInput;
  return staticPath && path.resolve(rootDir, staticPath);
}

function getHttps(cosmosConfigInput: CosmosConfigInput) {
  const { https = false } = cosmosConfigInput;
  return https;
}

function getHttpsOptions(
  cosmosConfigInput: CosmosConfigInput,
  rootDir: string
) {
  if (
    cosmosConfigInput.httpsOptions?.keyPath &&
    cosmosConfigInput.httpsOptions?.certPath
  ) {
    return {
      keyPath: path.resolve(rootDir, cosmosConfigInput.httpsOptions.keyPath),
      certPath: path.resolve(rootDir, cosmosConfigInput.httpsOptions.certPath),
    };
  }
  return null;
}

function getIgnore({ ignore = [] }: CosmosConfigInput) {
  return ['**/node_modules/**', ...ignore];
}

function getFixturesDir({ fixturesDir = '__fixtures__' }: CosmosConfigInput) {
  return fixturesDir;
}

function getFixtureFileSuffix({
  fixtureFileSuffix = 'fixture',
}: CosmosConfigInput) {
  return fixtureFileSuffix;
}

function getWatchDirs(cosmosConfigInput: CosmosConfigInput, rootDir: string) {
  const { watchDirs = ['.'] } = cosmosConfigInput;
  return watchDirs.map(watchDir => path.resolve(rootDir, watchDir));
}

function getExposeImports(
  cosmosConfigInput: CosmosConfigInput,
  rootDir: string
) {
  const cliArgs = getCliArgs();
  if (typeof cliArgs.exposeImports === 'boolean') {
    return cliArgs.exposeImports;
  } else if (typeof cliArgs.exposeImports === 'string') {
    return path.resolve(rootDir, cliArgs.exposeImports);
  }

  const { exposeImports = false } = cosmosConfigInput;
  return typeof exposeImports === 'string'
    ? path.resolve(rootDir, exposeImports)
    : exposeImports;
}

function getHostname({ hostname = null }: CosmosConfigInput) {
  return hostname;
}

function getPort(cosmosConfigInput: CosmosConfigInput) {
  const cliArgs = getCliArgs();
  if (typeof cliArgs.port === 'number') {
    return cliArgs.port;
  }

  const { port = 5000 } = cosmosConfigInput;
  return port;
}

function getPortRetries({ portRetries = 10 }: CosmosConfigInput) {
  return portRetries;
}

function getGlobalImports(
  cosmosConfigInput: CosmosConfigInput,
  rootDir: string
) {
  const { globalImports = [] } = cosmosConfigInput;
  return globalImports.map(globalImport => resolveLoose(rootDir, globalImport));
}

function getPlugins(cosmosConfigInput: CosmosConfigInput, rootDir: string) {
  const { plugins = [] } = cosmosConfigInput;
  return plugins.map(plugin => resolveLoose(rootDir, plugin));
}

function getDomConfig(cosmosConfigInput: CosmosDomConfigInput) {
  const { containerQuerySelector = null } = cosmosConfigInput;
  return {
    containerQuerySelector,
  };
}

function getLazy(cosmosConfigInput: CosmosConfigInput) {
  const cliArgs = getCliArgs();
  if (typeof cliArgs.lazy === 'boolean') {
    return cliArgs.lazy;
  }

  const { lazy = false } = cosmosConfigInput;
  return lazy;
}
