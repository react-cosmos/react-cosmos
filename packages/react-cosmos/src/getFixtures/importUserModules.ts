import {
  ByPath,
  ReactDecoratorModule,
  ReactFixtureModule,
} from 'react-cosmos-core';
import { CosmosConfig } from '../cosmosConfig/types.js';
import { findUserModulePaths } from '../userModules/findUserModulePaths.js';
import { importKeyPath } from '../userModules/shared.js';

type UserModules = {
  fixtures: ByPath<ReactFixtureModule>;
  decorators: ByPath<ReactDecoratorModule>;
};

export async function importUserModules({
  rootDir,
  fixturesDir,
  fixtureFileSuffix,
  ignore,
}: CosmosConfig): Promise<UserModules> {
  const { fixturePaths, decoratorPaths } = await findUserModulePaths({
    rootDir,
    fixturesDir,
    fixtureFileSuffix,
    ignore,
  });
  return {
    fixtures: await importModules(fixturePaths, rootDir),
    decorators: await importModules(decoratorPaths, rootDir),
  };
}

async function importModules<T>(paths: string[], rootDir: string) {
  const modules = await Promise.all(
    paths.map(async p => {
      const relPath = importKeyPath(p, rootDir);
      return { relPath, module: await import(p) };
    })
  );

  return modules.reduce(
    (acc: Record<string, T>, { relPath, module }) => ({
      ...acc,
      [relPath]: module,
    }),
    {}
  );
}
