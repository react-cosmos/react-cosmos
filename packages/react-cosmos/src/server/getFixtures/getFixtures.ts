import path from 'path';
import { ReactElement } from 'react';
import {
  createFixtureTree,
  FixtureId,
  flattenFixtureTree,
  getDecoratedFixtureElement,
  getFixtureListFromExports,
  getSortedDecoratorsForFixturePath,
  ReactDecorator,
  ReactDecorators,
  ReactFixture,
  ReactFixtureMap,
  stringifyPlaygroundUrlQuery,
  stringifyRendererUrlQuery,
} from 'react-cosmos-core';
import url from 'url';
import { CosmosConfig } from '../cosmosConfig/types.js';
import { RENDERER_FILENAME } from '../shared/playgroundHtml.js';
import { getUserModules } from '../userDeps/getUserModules.js';

export type FixtureApi = {
  absoluteFilePath: string;
  fileName: string;
  getElement: () => ReactElement;
  name: string | null;
  parents: string[];
  playgroundUrl: string;
  relativeFilePath: string;
  rendererUrl: string;
  treePath: string[];
};

export function getFixtures(cosmosConfig: CosmosConfig) {
  const { fixturesDir, fixtureFileSuffix, rootDir } = cosmosConfig;

  const fixtureInfo: FixtureApi[] = [];
  const { fixtures, decorators } = getUserModules(cosmosConfig);
  const fixtureList = getFixtureListFromExports(fixtures);
  const fixtureTree = createFixtureTree({
    fixtures: fixtureList,
    fixturesDir,
    fixtureFileSuffix,
  });
  const flatFixtureTree = flattenFixtureTree(fixtureTree);
  flatFixtureTree.forEach(({ fileName, fixtureId, name, parents }) => {
    const fixtureExport = fixtures[fixtureId.path];
    const fixture: ReactFixture =
      fixtureId.name === undefined
        ? (fixtureExport as ReactFixture)
        : (fixtureExport as ReactFixtureMap)[fixtureId.name];

    const treePath = [...parents, fileName];
    if (name) treePath.push(name);

    fixtureInfo.push({
      absoluteFilePath: path.join(rootDir, fixtureId.path),
      fileName,
      getElement: createFixtureElementGetter(
        fixture,
        fixtureId.path,
        decorators
      ),
      name,
      parents,
      playgroundUrl: getPlaygroundUrl(cosmosConfig, fixtureId),
      relativeFilePath: fixtureId.path,
      rendererUrl: getRendererUrl(cosmosConfig, fixtureId),
      treePath,
    });
  });

  return fixtureInfo;
}

function getPlaygroundUrl(cosmosConfig: CosmosConfig, fixtureId: FixtureId) {
  const host = getPlaygroundHost(cosmosConfig);
  const query = stringifyPlaygroundUrlQuery({ fixtureId });
  return `${host}/?${query}`;
}

function getRendererUrl(cosmosConfig: CosmosConfig, fixtureId: FixtureId) {
  const { publicUrl, experimentalRendererUrl } = cosmosConfig;
  const query = stringifyRendererUrlQuery({ _fixtureId: fixtureId });
  if (experimentalRendererUrl) return `${experimentalRendererUrl}?${query}`;

  const host = getPlaygroundHost(cosmosConfig);
  const urlPath = url.resolve(publicUrl, RENDERER_FILENAME);
  return `${host}${urlPath}?${query}`;
}

function getPlaygroundHost({ hostname, port, https }: CosmosConfig) {
  return `${https ? 'https' : 'http'}://${hostname || 'localhost'}:${port}`;
}

function createFixtureElementGetter(
  fixture: ReactFixture,
  fixturePath: string,
  decoratorsByPath: ReactDecorators
): () => ReactElement {
  const decorators: ReactDecorator[] = getSortedDecoratorsForFixturePath(
    fixturePath,
    decoratorsByPath
  );
  return () =>
    getDecoratedFixtureElement(fixture, decorators, {
      fixtureState: {},
      setFixtureState: () => {},
      onErrorReset: () => {},
    });
}
