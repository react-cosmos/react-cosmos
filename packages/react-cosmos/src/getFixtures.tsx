import React from 'react';
import { getDecoratedFixtureElement } from 'react-cosmos-shared2/FixtureLoader';
import {
  getSortedDecoratorsForFixturePath,
  isMultiFixture,
  ReactDecorator,
  ReactDecorators,
  ReactFixture,
  ReactFixtureMap,
} from 'react-cosmos-shared2/react';
import { FixtureId } from 'react-cosmos-shared2/renderer';
import { CosmosConfig } from './config';
import { getUserModules } from './shared/userDeps';

type Args = {
  cosmosConfig: CosmosConfig;
};

type RenderableFixture = {
  fixtureId: FixtureId;
  getElement: () => React.ReactElement<any>;
};

export const getFixtures = async (args: Args) => getFixturesSync(args);

export function getFixturesSync({ cosmosConfig }: Args) {
  const { fixtures, decorators } = getUserModules(cosmosConfig);

  const result: RenderableFixture[] = [];
  Object.keys(fixtures).forEach(fixturePath => {
    const fixtureExport = fixtures[fixturePath];
    if (isMultiFixture(fixtureExport)) {
      // FIXME: Why does fixtureExport need to be cast as ReactFixtureMap when
      // the type predicate returned by isMultiFixture already ensures it?
      const multiFixtureExport: ReactFixtureMap = fixtureExport;
      Object.keys(fixtureExport).forEach(fixtureName => {
        const fixtureId = { path: fixturePath, name: fixtureName };
        result.push({
          fixtureId,
          getElement: createFixtureElementGetter(
            multiFixtureExport[fixtureName],
            fixturePath,
            decorators
          ),
        });
      });
    } else {
      const fixtureId = { path: fixturePath };
      result.push({
        fixtureId,
        getElement: createFixtureElementGetter(
          fixtureExport,
          fixturePath,
          decorators
        ),
      });
    }
  });

  return result;
}

function createFixtureElementGetter(
  fixture: ReactFixture,
  fixturePath: string,
  decoratorsByPath: ReactDecorators
): () => React.ReactElement<any> {
  const decorators: ReactDecorator[] = getSortedDecoratorsForFixturePath(
    fixturePath,
    decoratorsByPath
  );
  return () =>
    getDecoratedFixtureElement(fixture, decorators, {
      fixtureState: {},
      setFixtureState: () => {},
      onErrorReset: () => {},
    });
}
