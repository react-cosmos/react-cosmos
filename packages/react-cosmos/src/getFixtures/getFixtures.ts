import { mapValues } from 'lodash-es';
import path from 'path';
import { ReactElement } from 'react';
import {
  buildPlaygroundQueryString,
  ByPath,
  createFixtureTree,
  createRendererUrl,
  FixtureId,
  flattenFixtureTree,
  getFixtureFromExport,
  getFixtureListFromExports,
  getSortedDecoratorsForFixturePath,
  ReactDecorator,
  ReactFixture,
  ReactFixtureExport,
} from 'react-cosmos-core';
import { createFixtureNode, decorateFixture } from 'react-cosmos-renderer';
import { CosmosConfig } from '../cosmosConfig/types.js';
import { getPlaygroundUrl } from '../shared/playgroundUrl.js';
import { importUserModules } from './importUserModules.js';

export type FixtureApi = {
  absoluteFilePath: string;
  fileName: string;
  getElement: () => ReactElement;
  name: string | null;
  parents: string[];
  playgroundUrl: string;
  relativeFilePath: string;
  rendererUrl: string | null;
  treePath: string[];
};

type Options = {
  rendererUrl?: string;
};
export async function getFixtures(
  cosmosConfig: CosmosConfig,
  options: Options = {}
) {
  const { fixtures, decorators } = await importUserModules(cosmosConfig);
  const fixtureExports = mapValues(fixtures, f => f.default);
  const decoratorExports = mapValues(decorators, f => f.default);
  const result: FixtureApi[] = [];

  getFlatFixtureTree(cosmosConfig, fixtureExports).forEach(
    ({ fileName, fixtureId, name, parents }) => {
      const fixtureExport = fixtures[fixtureId.path].default;
      const fixtureOptions = fixtures[fixtureId.path].options ?? {};
      const fixture = getFixtureFromExport(fixtureExport, fixtureId.name);

      if (!fixture) {
        throw new Error(`Could not read fixture: ${JSON.stringify(fixtureId)}`);
      }

      const treePath = [...parents, fileName];
      if (name) treePath.push(name);

      result.push({
        absoluteFilePath: path.join(cosmosConfig.rootDir, fixtureId.path),
        fileName,
        getElement: createFixtureElementGetter(
          fixture,
          fixtureOptions,
          fixtureId.path,
          decoratorExports
        ),
        name,
        parents,
        playgroundUrl: getPlaygroundFixtureUrl(cosmosConfig, fixtureId),
        relativeFilePath: fixtureId.path,
        rendererUrl: options.rendererUrl
          ? createRendererUrl(options.rendererUrl, fixtureId)
          : null,
        treePath,
      });
    }
  );

  return result;
}

function getFlatFixtureTree(
  cosmosConfig: CosmosConfig,
  fixtures: ByPath<ReactFixtureExport>
) {
  const { fixturesDir, fixtureFileSuffix } = cosmosConfig;
  return flattenFixtureTree(
    createFixtureTree({
      fixtures: getFixtureListFromExports(fixtures),
      fixturesDir,
      fixtureFileSuffix,
    })
  );
}

function getPlaygroundFixtureUrl(
  cosmosConfig: CosmosConfig,
  fixtureId: FixtureId
) {
  const baseUrl = getPlaygroundUrl(cosmosConfig);
  const query = buildPlaygroundQueryString({ fixtureId });
  return `${baseUrl}/${query}`;
}

function createFixtureElementGetter(
  fixture: ReactFixture,
  fixtureOptions: {},
  fixturePath: string,
  decoratorsByPath: ByPath<ReactDecorator | ReactDecorator[]>
): () => ReactElement {
  const decorators = getSortedDecoratorsForFixturePath(
    fixturePath,
    decoratorsByPath
  );
  return () =>
    decorateFixture(
      createFixtureNode(fixture),
      fixtureOptions,
      decorators.flat()
    );
}
