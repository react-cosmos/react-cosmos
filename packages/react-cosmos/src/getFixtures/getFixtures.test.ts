import { render } from '@testing-library/react';
import path from 'node:path';
import { createCosmosConfig } from '../cosmosConfig/createCosmosConfig.js';
import { getFixtures } from './getFixtures.js';

const rootDir = path.join(__dirname, '../../../../examples/webpack');

it('renders fixture elements', async () => {
  const cosmosConfig = createCosmosConfig(rootDir, {
    ignore: ['**/*.mdx'],
  });

  const fixures = await getFixtures(cosmosConfig, {
    rendererUrl: 'http://localhost:5000/renderer.html',
  });

  function testFixtureElement(relPath: string, name: string | null = null) {
    const match = fixures.find(
      f => f.relativeFilePath === relPath && f.name === name
    );
    const renderer = render(match!.getElement());
    expect(renderer.container).toMatchSnapshot();
  }

  testFixtureElement('src/CounterButton.fixture.tsx');
  testFixtureElement('src/Counter.fixture.tsx', 'large number');
  testFixtureElement('src/WelcomeMessage/WelcomeMessage.fixture.tsx');
});

it('returns fixture info', async () => {
  const cosmosConfig = createCosmosConfig(rootDir, {
    ignore: ['**/*.mdx'],
  });

  const fixtures = await getFixtures(cosmosConfig, {
    rendererUrl: 'http://localhost:5000/renderer.html',
  });

  expect(fixtures).toEqual([
    {
      absoluteFilePath: path.join(
        rootDir,
        'src/__fixtures__/controls/Inputs Panel.tsx'
      ),
      fileName: 'Inputs Panel',
      getElement: expect.any(Function),
      name: null,
      parents: ['controls'],
      relativeFilePath: 'src/__fixtures__/controls/Inputs Panel.tsx',
      rendererUrl:
        'http://localhost:5000/renderer.html?fixtureId=%7B%22path%22%3A%22src%2F__fixtures__%2Fcontrols%2FInputs+Panel.tsx%22%7D',
      playgroundUrl:
        'http://localhost:5000/?fixtureId=%7B%22path%22%3A%22src%2F__fixtures__%2Fcontrols%2FInputs+Panel.tsx%22%7D',
      treePath: ['controls', 'Inputs Panel'],
    },
    {
      absoluteFilePath: path.join(
        rootDir,
        'src/__fixtures__/controls/Props Panel.tsx'
      ),
      fileName: 'Props Panel',
      getElement: expect.any(Function),
      name: null,
      parents: ['controls'],
      relativeFilePath: 'src/__fixtures__/controls/Props Panel.tsx',
      rendererUrl:
        'http://localhost:5000/renderer.html?fixtureId=%7B%22path%22%3A%22src%2F__fixtures__%2Fcontrols%2FProps+Panel.tsx%22%7D',
      playgroundUrl:
        'http://localhost:5000/?fixtureId=%7B%22path%22%3A%22src%2F__fixtures__%2Fcontrols%2FProps+Panel.tsx%22%7D',
      treePath: ['controls', 'Props Panel'],
    },
    {
      absoluteFilePath: path.join(rootDir, 'src/Counter.fixture.tsx'),
      fileName: 'Counter',
      getElement: expect.any(Function),
      name: 'default',
      parents: [],
      playgroundUrl:
        'http://localhost:5000/?fixtureId=%7B%22path%22%3A%22src%2FCounter.fixture.tsx%22%2C%22name%22%3A%22default%22%7D',
      relativeFilePath: 'src/Counter.fixture.tsx',
      rendererUrl:
        'http://localhost:5000/renderer.html?fixtureId=%7B%22path%22%3A%22src%2FCounter.fixture.tsx%22%2C%22name%22%3A%22default%22%7D',
      treePath: ['Counter', 'default'],
    },
    {
      absoluteFilePath: path.join(rootDir, 'src/Counter.fixture.tsx'),
      fileName: 'Counter',
      getElement: expect.any(Function),
      name: 'small number',
      parents: [],
      playgroundUrl:
        'http://localhost:5000/?fixtureId=%7B%22path%22%3A%22src%2FCounter.fixture.tsx%22%2C%22name%22%3A%22small+number%22%7D',
      relativeFilePath: 'src/Counter.fixture.tsx',
      rendererUrl:
        'http://localhost:5000/renderer.html?fixtureId=%7B%22path%22%3A%22src%2FCounter.fixture.tsx%22%2C%22name%22%3A%22small+number%22%7D',
      treePath: ['Counter', 'small number'],
    },
    {
      absoluteFilePath: path.join(rootDir, 'src/Counter.fixture.tsx'),
      fileName: 'Counter',
      getElement: expect.any(Function),
      name: 'large number',
      parents: [],
      playgroundUrl:
        'http://localhost:5000/?fixtureId=%7B%22path%22%3A%22src%2FCounter.fixture.tsx%22%2C%22name%22%3A%22large+number%22%7D',
      relativeFilePath: 'src/Counter.fixture.tsx',
      rendererUrl:
        'http://localhost:5000/renderer.html?fixtureId=%7B%22path%22%3A%22src%2FCounter.fixture.tsx%22%2C%22name%22%3A%22large+number%22%7D',
      treePath: ['Counter', 'large number'],
    },
    {
      absoluteFilePath: path.join(rootDir, 'src/CounterButton.fixture.tsx'),
      fileName: 'CounterButton',
      getElement: expect.any(Function),
      name: null,
      parents: [],
      playgroundUrl:
        'http://localhost:5000/?fixtureId=%7B%22path%22%3A%22src%2FCounterButton.fixture.tsx%22%7D',
      relativeFilePath: 'src/CounterButton.fixture.tsx',
      rendererUrl:
        'http://localhost:5000/renderer.html?fixtureId=%7B%22path%22%3A%22src%2FCounterButton.fixture.tsx%22%7D',
      treePath: ['CounterButton'],
    },
    {
      absoluteFilePath: path.join(
        rootDir,
        'src/WelcomeMessage/WelcomeMessage.fixture.tsx'
      ),
      fileName: 'WelcomeMessage',
      getElement: expect.any(Function),
      name: null,
      parents: [],
      playgroundUrl:
        'http://localhost:5000/?fixtureId=%7B%22path%22%3A%22src%2FWelcomeMessage%2FWelcomeMessage.fixture.tsx%22%7D',
      relativeFilePath: 'src/WelcomeMessage/WelcomeMessage.fixture.tsx',
      rendererUrl:
        'http://localhost:5000/renderer.html?fixtureId=%7B%22path%22%3A%22src%2FWelcomeMessage%2FWelcomeMessage.fixture.tsx%22%7D',
      treePath: ['WelcomeMessage'],
    },
  ]);
});
