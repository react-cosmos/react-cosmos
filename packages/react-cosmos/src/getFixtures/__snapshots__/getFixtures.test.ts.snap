// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders fixture elements 1`] = `
<div>
  <button
    title="Click to increment"
  >
    0
     
    times
  </button>
</div>
`;

exports[`renders fixture elements 2`] = `
<div>
  <button
    title="Click to increment"
  >
    555555555
     
    times
  </button>
</div>
`;

exports[`renders fixture elements 3`] = `
<div>
  <div
    style="border: 2px dashed #aaa; padding: 0px 16px; border-radius: 8px;"
  >
    <h1>
      Hi
       
      <strong>
        Maggie
      </strong>
      !
    </h1>
  </div>
</div>
`;
